diff --git a/drivers/media/platform/msm/camera/camera_kt/drivers/cam_req_mgr/cam_mem_mgr.c b/drivers/media/platform/msm/camera/camera_kt/drivers/cam_req_mgr/cam_mem_mgr.c
--- a/drivers/media/platform/msm/camera/camera_kt/drivers/cam_req_mgr/cam_mem_mgr.c
+++ b/drivers/media/platform/msm/camera/camera_kt/drivers/cam_req_mgr/cam_mem_mgr.c
@@ -724,12 +724,30 @@ static int cam_mem_util_get_dma_buf(size_t len,
 
 	if (tbl.alloc_profile_enable) {
 		CAM_GET_TIMESTAMP(ts2);
 		CAM_GET_TIMESTAMP_DIFF_IN_MICRO(ts1, ts2, microsec);
 		trace_cam_log_event("IONAllocProfile", "size and time in micro",
 			len, microsec);
 	}
 #endif
+#if !defined(CONFIG_CAM_ION_SUPPORT)
+#if !defined(USE_DOWNSTREAM_DMA_BUFF)
+	CAM_DBG(CAM_MEM, "Using cmm_alloc_buffer for allocation, len=%zu, flags=0x%x",
+		len, cam_flags);
+	*buf = cmm_alloc_buffer(len, cam_flags);
+	if (IS_ERR_OR_NULL(*buf)) {
+		rc = IS_ERR(*buf) ? PTR_ERR(*buf) : -ENOMEM;
+		CAM_ERR(CAM_MEM,
+			"Failed in allocating from cmm buffer manager, len=%zu, flags=0x%x, rc=%d",
+			len, cam_flags, rc);
+		*buf = NULL;
+		return rc;
+	}
+#else
+	CAM_ERR(CAM_MEM, "No camera memory allocator is available");
+	return -EINVAL;
+#endif
+#endif
 
 	return rc;
 }
 #endif
