diff --git a/drivers/media/platform/msm/camera/camera_kt/drivers/cam_smmu/cam_smmu_api.c b/drivers/media/platform/msm/camera/camera_kt/drivers/cam_smmu/cam_smmu_api.c
--- a/drivers/media/platform/msm/camera/camera_kt/drivers/cam_smmu/cam_smmu_api.c
+++ b/drivers/media/platform/msm/camera/camera_kt/drivers/cam_smmu/cam_smmu_api.c
@@ -1912,7 +1912,6 @@ err_detach:
 	dma_buf_detach(secheap_buf->buf,
 		secheap_buf->attach);
 err_put:
-	dma_buf_put(secheap_buf->buf);
 err_out:
 	mutex_unlock(&iommu_cb_set.cb_info[idx].lock);
 	return rc;
@@ -1962,7 +1961,6 @@ int cam_smmu_release_sec_heap(int32_t smmu_hdl)
 
 	dma_buf_unmap_attachment(secheap_buf->attach,
 		secheap_buf->table, DMA_BIDIRECTIONAL);
 	dma_buf_detach(secheap_buf->buf, secheap_buf->attach);
-	dma_buf_put(secheap_buf->buf);
 	iommu_cb_set.cb_info[idx].is_secheap_allocated = false;
 	mutex_unlock(&iommu_cb_set.cb_info[idx].lock);
 
